server/protocols/jmap/src/main/java/org/apache/james/jmap/model/AttachmentAccessToken.java

```diff
@@ -0,0 +1,128 @@
+
+package org.apache.james.jmap.model;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class AttachmentAccessToken implements SignedExpiringToken {
+
+    public static final String SEPARATOR = "_";
+
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    public static class Builder {
+        private String blobId;
+        private ZonedDateTime expirationDate;
+        private String signature;
+
+        private Builder() {}
+        
+        public Builder blobId(String blobId) {
+            this.blobId = blobId;
+            return this;
+        }
+
+        public Builder expirationDate(ZonedDateTime expirationDate) {
+            this.expirationDate = expirationDate;
+            return this;
+        }
+
+        public Builder signature(String signature) {
+            this.signature = signature;
+            return this;
+        }
+
+        public AttachmentAccessToken build() {
+            Preconditions.checkNotNull(blobId);
+            Preconditions.checkArgument(! blobId.isEmpty());
+            Preconditions.checkNotNull(expirationDate);
+            Preconditions.checkNotNull(signature);
+            return new AttachmentAccessToken(blobId, expirationDate, signature);
+        }
+    }
+    
+    private final String blobId;
+    private final ZonedDateTime expirationDate;
+    private final String signature;
+
+    @VisibleForTesting
+    AttachmentAccessToken(String blobId, ZonedDateTime expirationDate, String signature) {
+        this.blobId = blobId;
+        this.expirationDate = expirationDate;
+        this.signature = signature;
+    }
+
+    public String getBlobId() {
+        return blobId;
+    }
+
+    @Override
+    public ZonedDateTime getExpirationDate() {
+        return expirationDate;
+    }
+
+    @Override
+    public String getSignature() {
+        return signature;
+    }
+
+    public String serialize() {
+        return getContent()
+            + SEPARATOR
+            + signature;
+    }
+    
+    @Override
+    public String getContent() {
+        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(expirationDate);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        AttachmentAccessToken attachmentAccessToken = (AttachmentAccessToken) other;
+        return Objects.equals(blobId, attachmentAccessToken.blobId)
+            && expirationDate.isEqual(attachmentAccessToken.expirationDate)
+            && Objects.equals(signature, attachmentAccessToken.signature);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(blobId, expirationDate, signature);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Auto generated by intelliJ ?

I guess we uses toStringHelper from MoreObjects every where else...